xonsh/main.py

```diff
@@ -224,8 +224,8 @@ def main(argv=None):
         # run a script contained in a file
         path = os.path.abspath(os.path.expanduser(args.file))
         if os.path.isfile(path):
-            sys.argv = args.args
-            env['ARGS'] = [args.file] + args.args
+            sys.argv = [args.file] + args.args
+            env['ARGS'] = sys.argv[:]  # $ARGS is not sys.argv
```

Comment:
in my point of view $ARGS envvar is  an alias/shortcut for sys.argv so you can get the arguments in xonsh scripts without having to `import sys`. The explicit thing here imo is that both of these names point to the same object. One bug that may come up involves `argparser.ArgumentParser.parse_args()` which reads the arguments from `sys.stdin`, if a xonsh script mutates the `$ARGS` envvar the changes wont be visible to `parse_args()`